File: src/main/java/com/epam/eng/pattern/adapter/Plug.java
package com.epam.eng.pattern.adapter;

public class Plug {

    class AmericanPlug {
        public void insertIntoAmericanSocket() {
            System.out.println("Inserting into American socket.");
        }
    }

    class EuropeanPlug {
        public void insertIntoEuropeanSocket() {
            System.out.println("Inserting into European socket.");
        }
    }


    public static void main(String[] args) {
        /* TODO: Use the AmericanPlug in a European socket */
    }
}

File: ts/adapter/Plug.ts
class AmericanPlug {
  insertIntoAmericanSocket(): void {
    console.log("Inserting into American socket.");
  }
}

class EuropeanPlug {
  insertIntoEuropeanSocket(): void {
    console.log("Inserting into European socket.");
  }
}

class AmericanToEuropeanPlugAdapter extends EuropeanPlug {
  private americanPlug: AmericanPlug;
  constructor(americanPlug: AmericanPlug) {
    super();
    this.americanPlug = americanPlug;
  }

  insertIntoEuropeanSocket(): void {
    this.americanPlug.insertIntoAmericanSocket();
  }
}

const myAmericanPlug = new AmericanPlug();
const adapter = new AmericanToEuropeanPlugAdapter(myAmericanPlug);
adapter.insertIntoEuropeanSocket();
